Add initial implementation.

lrhn opened the pull request.

mit-mit committed:
Repo config

lrhn committed:
Merge branch 'master' of https://github.com/dart-lang/characters

lrhn committed:
Initial implementation.

lrhn committed:
Add test runs to Travis.

kevmoo commented on .gitignore:
```diff
@@ -0,0 +1,10 @@
+.buildlog
```
Less is more here: https://dart.dev/guides/libraries/private-files

lrhn replied:
I just copied this from some other repo. :)

I'll just go wtih
* dart-tool/
* .packages
* pubspec.lock

I'm not "build"-ing anything.

kevmoo commented on README.md:
```diff
@@ -1,4 +1,10 @@
 # characters
```
Remove superfluous title in readme. Redundant on both github page and pub site

lrhn replied:
Done.

kevmoo commented on travis.yml:
```diff
@@ -0,0 +1,4 @@
+language: dart
```
Why this AND `.travis.yml`

lrhn replied:
No reason. Removing.

kevmoo commented on .travis.yml:
```diff
@@ -1,4 +1,6 @@
 language: dart
 dart_task:
+- test: --platform vm
```
Consider combining these? Booting up and running `pub get` costs a minute for each run.

lrhn replied:
What would the syntax of that be? I can't see anything useful on https://docs.travis-ci.com/user/languages/dart/

kevmoo replied:
`--platform vm,chrome`

kevmoo commented on .travis.yml:
```diff
@@ -1,4 +1,6 @@
+- test: --platform vm
+- test: --platform chrome
 - dartanalyzer
 - dartfmt
```
Consider adding

```yaml
dart
- dev
- 2.4.0
```

Good to test on oldest supported SDK

lrhn replied:
Done

lrhn committed:
Remove `CharacterBoundaries`.

The functionality is available through the `Character` class, and it does not carry its own weight
as a top-level function (which was, for some reason, capitalized).

lrhn committed:
Remove trailing spaces.

lrhn commented:
Should probably mention in readme that this is *experimental*.

kevmoo commented on .travis.yml:
```diff
@@ -1,4 +1,9 @@
+- test: --platform vm
+- test: --platform chrome
 - dartanalyzer
 - dartfmt
```
Also add

```yaml
# Only building master means that we don't run two builds for each pull request.
branches:
  only: [master]

cache:
 directories:
   - $HOME/.pub-cache
```

lrhn committed:
Update .travis.yml

lrhn committed:
Rename top-level library to "characters.dart".
